email-encryption-demo/src/com/axonivy/market/emailencryption/service/EncryptedEmailSender.java

```diff
@@ -0,0 +1,211 @@
+package com.axonivy.market.emailencryption.service;
+
+import java.io.UnsupportedEncodingException;
+import java.security.Security;
+import java.security.cert.X509Certificate;
+import java.util.Properties;
+import java.util.concurrent.Callable;
+
+import javax.activation.CommandMap;
+import javax.activation.DataHandler;
+import javax.activation.DataSource;
+import javax.activation.FileDataSource;
+import javax.activation.MailcapCommandMap;
+import javax.mail.Address;
+import javax.mail.Authenticator;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+
+import org.apache.commons.lang3.StringUtils;
+import org.bouncycastle.cms.CMSAlgorithm;
+import org.bouncycastle.cms.jcajce.JceCMSContentEncryptorBuilder;
+import org.bouncycastle.cms.jcajce.JceKeyTransRecipientInfoGenerator;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.mail.smime.SMIMEEnvelopedGenerator;
+import org.bouncycastle.mail.smime.SMIMEException;
+import org.bouncycastle.operator.OutputEncryptor;
+import org.eclipse.core.resources.IProject;
+
+import ch.ivyteam.ivy.environment.Ivy;
+import ch.ivyteam.ivy.mail.MailClientConfig;
+import ch.ivyteam.ivy.mail.MailClientConfigProvider;
+import ch.ivyteam.ivy.mail.MailConstants.EmailEncryption;
+import ch.ivyteam.ivy.project.IIvyProject;
+import ch.ivyteam.ivy.scripting.objects.File;
+import ch.ivyteam.ivy.security.SecurityManagerFactory;
+import ch.ivyteam.util.IvyException;
+
+/**
+ * Service to send encrypted mails.
+ * 
+ */
+public class EncryptedEmailSender {

```

Comment:
this service and it's surrounding APIs seems to be independent from the 'demo'.
I'd isolate it in a product, so one can integrate it into a project-workspace, without actually copying files from a demo.